File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) 
    {
        int lb=0;
        int ub=nums.size()-1;
        while(lb<=ub)
        {
            int mid=(ub+lb)/2;
            if(nums[mid]==target)
            {
                return mid;
            }
            else if(target<nums[mid])
                ub=mid-1;
            else
                lb=mid+1;
        }
        return -1;
    }
};

File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, u = nums.length - 1, mid = 0;
        while(l<=u)
        {
            mid = (l+u)/2;
            if(nums[mid] == target)
            {
                return mid;
            }
            else if(nums[mid] > target)
            {
                u = mid - 1;
            }
            else
            {
                l = mid + 1;
            }
        }
        return -1;
    }
}
//nb=